src/api/composition/CompositionCollection.js

```diff
@@ -20,46 +20,68 @@
  * at runtime from the About dialog for additional information.
  *****************************************************************************/
 
-define([
-    'lodash'
-], function (
-    _
-) {
+/**
+ * @typedef {import('../objects/ObjectAPI').DomainObject} DomainObject
+ */
+
+/**
+ * @typedef {import('./CompositionAPI').default} CompositionAPI
+ */
+
+/**
+ * @typedef {import('../../../openmct').OpenMCT} OpenMCT
+ */
+
+/**
+ * @typedef {object} ListenerMap
+ * @property {Array.<any>} add
+ * @property {Array.<any>} remove
+ * @property {Array.<any>} load
+ * @property {Array.<any>} reorder
+ */
+
+/**
+ * A CompositionCollection represents the list of domain objects contained
+ * by another domain object. It provides methods for loading this
+ * list asynchronously, modifying this list, and listening for changes to
+ * this list.
+ *
+ * Usage:
+ * ```javascript
+ *  var myViewComposition = MCT.composition.get(myViewObject);
+ *  myViewComposition.on('add', addObjectToView);
+ *  myViewComposition.on('remove', removeObjectFromView);
+ *  myViewComposition.load(); // will trigger `add` for all loaded objects.
+ *  ```
+ *
+ * @class
+ */
+export default class CompositionCollection {
     /**
-     * A CompositionCollection represents the list of domain objects contained
-     * by another domain object. It provides methods for loading this
-     * list asynchronously, modifying this list, and listening for changes to
-     * this list.
-     *
-     * Usage:
-     * ```javascript
-     *  var myViewComposition = MCT.composition.get(myViewObject);
-     *  myViewComposition.on('add', addObjectToView);
-     *  myViewComposition.on('remove', removeObjectFromView);
-     *  myViewComposition.load(); // will trigger `add` for all loaded objects.
-     *  ```
-     *
-     * @interface CompositionCollection
-     * @param {module:openmct.DomainObject} domainObject the domain object
+     * @constructor
+     * @param {DomainObject} domainObject the domain object
      *        whose composition will be contained
-     * @param {module:openmct.CompositionProvider} provider the provider
+     * @param {import('./CompositionProvider').default} provider the provider
      *        to use to retrieve other domain objects
-     * @param {module:openmct.CompositionAPI} api the composition API, for
+     * @param {OpenMCT} publicAPI the composition API, for
      *        policy checks
-     * @memberof module:openmct
      */
-    function CompositionCollection(domainObject, provider, publicAPI) {
+    constructor(domainObject, provider, publicAPI) {
+        /** @type {DomainObject} */
```

Comment:
Basically all of these member variables should be private. Because this is public API and we don't want people building dependencies on internal structure that may change let's make them #private attributes.

Reply:
Done, except for `onProviderAdd` and `onProviderRemove` as they are used in callbacks